src/types/index.ts

```diff
@@ -243,3 +243,160 @@ export interface ERC20 extends Contract {
+    >
+
+    numTokens(): TransactionObject<string>
+
+    lastCreditBatchId(arg0: string, arg1: string): TransactionObject<string>
+
+    latestSolution(): TransactionObject<{
+      batchId: string
+      solutionSubmitter: string
+      feeReward: string
+      objectiveValue: string
+      0: string
+      1: string
+      2: string
+      3: string
+    }>
+
+    getBalance(user: string, token: string): TransactionObject<string, [string, string]>
+
+    getCurrentBatchId(): TransactionObject<string>
+
+    requestFutureWithdraw(
+      token: string,
+      amount: number | string | BN,
+      batchId: number | string | BN,
+    ): TransactionObject<void, [string, number | string | BN, number | string | BN]>
+
+    hasValidWithdrawRequest(user: string, token: string): TransactionObject<boolean, [string, string]>
+
+    MAX_TOKENS(): TransactionObject<string>
+
+    getPendingDepositBatchNumber(user: string, token: string): TransactionObject<string, [string, string]>
+
+    withdraw(user: string, token: string): TransactionObject<void, [string, string]>
+
+    MAX_TOUCHED_ORDERS(): TransactionObject<string>
+
+    addToken(token: string): TransactionObject<void, [string]>
+
+    placeValidFromOrders(
+      buyTokens: (number | string | BN)[],
+      sellTokens: (number | string | BN)[],
+      validFroms: (number | string | BN)[],
+      validUntils: (number | string | BN)[],
+      buyAmounts: (number | string | BN)[],
+      sellAmounts: (number | string | BN)[],
+    ): TransactionObject<
+      string[],
+      [
+        (number | string | BN)[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+      ]
+    >
+
+    placeOrder(
+      buyToken: number | string | BN,
+      sellToken: number | string | BN,
+      validUntil: number | string | BN,
+      buyAmount: number | string | BN,
+      sellAmount: number | string | BN,
+    ): TransactionObject<
+      string,
+      [number | string | BN, number | string | BN, number | string | BN, number | string | BN, number | string | BN]
+    >
+
+    cancelOrder(ids: (number | string | BN)[]): TransactionObject<void, [(number | string | BN)[]]>
+
+    freeStorageOfOrder(ids: (number | string | BN)[]): TransactionObject<void, [(number | string | BN)[]]>
+
+    submitSolution(
+      batchIndex: number | string | BN,
+      claimedObjectiveValue: number | string | BN,
+      owners: string[],
+      orderIds: (number | string | BN)[],
+      volumes: (number | string | BN)[],
+      prices: (number | string | BN)[],
+      tokenIdsForPrice: (number | string | BN)[],
+    ): TransactionObject<
+      string,
+      [
+        number | string | BN,
+        number | string | BN,
+        string[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+        (number | string | BN)[],
+      ]
+    >
```

Comment:
Do we need to have these methods mapped in the fronted?

Reply:
They are only mapped as far as types go. So that we know we have them on `Contract.methods`